radix-engine-tests/tests/system/crypto_utils.rs

```diff
@@ -659,3 +678,257 @@ fn test_crypto_scrypto_bls12381_v1_fast_aggregate_verify_costing() {
+
+    // Act
+    let msg2_verify: bool = get_output!(crypto_scrypto_ed25519_verify(
+        &mut ledger,
+        package_address,
+        msg2,
+        pk,
+        msg1_signature,
+    ));
+
+    // Assert
+    assert!(!msg2_verify);
+}
+
+#[test]
+fn test_crypto_scrypto_verify_secp256k1_ecdsa() {
+    // Arrange
+    let mut ledger = LedgerSimulatorBuilder::new().build();
+
+    let package_address = ledger.publish_package_simple(PackageLoader::get("crypto_scrypto_v2"));
+
+    let hash1 = hash("Test");
+    let pk = "0279be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798";
+    let hash1_signature = "00eb8dcd5bb841430dd0a6f45565a1b8bdb4a204eb868832cd006f963a89a662813ab844a542fcdbfda4086a83fbbde516214113051b9c8e42a206c98d564d7122";
+
+    let pk = Secp256k1PublicKey::from_str(pk).unwrap();
+    let hash1_signature = Secp256k1Signature::from_str(hash1_signature).unwrap();
+    // Act
+    let msg1_verify: bool = get_output!(crypto_scrypto_secp256k1_ecdsa_verify(
+        &mut ledger,
+        package_address,
+        hash1,
+        pk,
+        hash1_signature,
+    ));
+
+    // Assert
+    assert!(msg1_verify);
+
+    // Arrange
+    let hash2 = hash("ExpectFailureTest");
+
+    // Act
+    let msg2_verify: bool = get_output!(crypto_scrypto_secp256k1_ecdsa_verify(
+        &mut ledger,
+        package_address,
+        hash2,
+        pk,
+        hash1_signature,
+    ));
+
+    // Assert
+    assert!(!msg2_verify);
+}
+
+#[test]
+fn test_crypto_scrypto_blake2b_256_costing() {
+    let mut ledger = LedgerSimulatorBuilder::new().build();
+
+    let package_address = ledger.publish_package_simple(PackageLoader::get("crypto_scrypto_v2"));
+
+    for size in [
+        100usize,
+        200,
+        500,
+        1024,
+        10 * 1024,
+        20 * 1024,
+        50 * 1024,
+        100 * 1024,
+        200 * 1024,
+        500 * 1024,
+        900 * 1024,
+    ] {
+        let data = vec![0u8; size];
+        let _hash = crypto_scrypto_blake_2b_256_hash(&mut ledger, package_address, data);
+    }
+}
+
+#[test]
+fn test_crypto_scrypto_verify_ed25519_costing() {
+    let mut ledger = LedgerSimulatorBuilder::new().build();
+
+    let package_address = ledger.publish_package_simple(PackageLoader::get("crypto_scrypto_v2"));
+
+    let secret_key = Ed25519PrivateKey::from_u64(1).unwrap();
+    let public_key = secret_key.public_key();
+
+    for size in [
+        100usize,
+        200,
+        500,
+        1024,
+        10 * 1024,
+        20 * 1024,
+        50 * 1024,
+        100 * 1024,
+        200 * 1024,
+        500 * 1024,
+        900 * 1024,
+    ] {
+        let data = vec![0u8; size];
+        let signature = secret_key.sign(&data);
+        let _ = crypto_scrypto_ed25519_verify(
+            &mut ledger,
+            package_address,
+            data,
+            public_key,
+            signature,
+        );
+    }
+}
+
+#[test]
+fn test_crypto_scrypto_verify_secp256k1_ecdsa_costing() {
```

Comment:
Out of interest - how is this test testing costing? It seems like more of a smoke test to check that it can work for a few different hashes?

Reply:
see above